node/src/binding/CanvasGradient.cc

```diff
@@ -0,0 +1,90 @@
+#include "CanvasGradient.h"
+#include "Context2D.h"
+#include "NodeBindingUtil.h"
+namespace NodeBinding
+{
+Napi::FunctionReference Gradient::constructor;
+Gradient::Gradient(const Napi::CallbackInfo &info) : Napi::ObjectWrap<Gradient>(info)
+{
+    if (info.Length() == 4)
+    {
+        this->mLinearGradientInfo = std::make_shared<LinearGradientInfo>();
+        this->mLinearGradientInfo->startX = info[0].As<Napi::Number>().FloatValue();
+        this->mLinearGradientInfo->startY = info[1].As<Napi::Number>().FloatValue();
+        this->mLinearGradientInfo->endX = info[2].As<Napi::Number>().FloatValue();
+        this->mLinearGradientInfo->endY = info[3].As<Napi::Number>().FloatValue();
+    }
+    else if (info.Length() == 6)
+    {
+        this->mRadialGradientInfo = std::make_shared<RadialGradientInfo>();
+        this->mRadialGradientInfo->startX = info[0].As<Napi::Number>().FloatValue();
+        this->mRadialGradientInfo->startY = info[1].As<Napi::Number>().FloatValue();
+        this->mRadialGradientInfo->startR = info[2].As<Napi::Number>().FloatValue();
+        this->mRadialGradientInfo->endX = info[3].As<Napi::Number>().FloatValue();
+        this->mRadialGradientInfo->endY = info[4].As<Napi::Number>().FloatValue();
+        this->mRadialGradientInfo->endR = info[5].As<Napi::Number>().FloatValue();
+    }
+    else
+    {
+        throwError(info, "wrong argment number");
+    }
+}
+void Gradient::Init(Napi::Env env)
+{
+    Napi::HandleScope scope(env);
+
+    Napi::Function func =
+        DefineClass(env,
+                    "gradient",
+                    {
+                        InstanceMethod("addColorStop", &Gradient::addColorStop),
+                    });
+    constructor = Napi::Persistent(func);
+    constructor.SuppressDestruct();
+}
+
+Napi::Object Gradient::NewInstance(Napi::Env env, const Napi::CallbackInfo &info)
+{
+    Napi::EscapableHandleScope scope(env);
+    if (info.Length() == 4)
+    {
+        Napi::Object obj = constructor.New({info[0], info[1], info[2], info[3]});
+        obj.Set("name", Napi::String::New(env, "linearGradient"));
+        return scope.Escape(napi_value(obj)).ToObject();
+    }
+    else if (info.Length() == 6)
+    {
+        Napi::Object obj = constructor.New({info[0], info[1], info[2], info[3], info[4], info[5]});
+        obj.Set("name", Napi::String::New(env, "radialGradient"));
+        return scope.Escape(napi_value(obj)).ToObject();
+    }
+    else
+    {
+        throwError(info, "wrong argument number");
```

Comment:
我们已经启用了 C++ 的 exception 的特性，是否可以直接构造 Error instance throw 不需要写后续的 dumb return？

Reply:
不太理解您的意思，我这里throwError是向nodejs层抛出一个exception，不是c++的exception，您的意思是？

Reply:
就是这里直接使用 C++ 关键字 throw，因为已经启用了 C++ 的 exception 特性，node-addon-api 会在函数调用栈上捕获 Napi::Error 抛到 JS 中。
```suggestion
        throw Error::New(env, "wrong argument number");
```

Reply:
明白，我这里尝试使用thrrow试试看